server/cmwell-cons/src/main/scala/Upgrade.scala

```diff
@@ -39,16 +39,28 @@ object Upgrade {
   }
 
   def runPostUpgradeActions(currentVersionF : Future[String], upgraded : String, hosts: GenSeq[String]): Future[Boolean]= {
-    def shouldRun(test : SemVersion, curr : SemVersion, upgrade : SemVersion) : Boolean = (test > curr) && (test <= upgrade)
+    /**
+      *  Note! According to semantic versions rules, what will be taken into account is only <major>.<minor>.<patch>
+      *    all test after + sign will not affect precedence!
+      *    Examples:  1.2.5  >  1.2.5
+      *       but     1.2.5  ==   1.2.5+167
+      */
+    def shouldRun(test : SemVer, curr : SemVer, upgrade : SemVer) : Boolean = (test > curr) && (test <= upgrade)
 
     currentVersionF.foreach{current =>
-      val currentVersion = Version(current)
-      val upgradedVersion = Version(upgraded)
+      val currentVersion = SemVer(current)
+      val upgradedVersion = SemVer(upgraded)
 
-      val relevantUpgradeFunctions = postUpgradeList.filter(f => shouldRun(Version(f.applyToVersion), currentVersion, upgradedVersion))
+      if (currentVersion.isEmpty || upgradedVersion.isEmpty)
+          error(s"Failed to parse the versions:\n ${
```

Comment:
According to your code, the "good" version will be displayed and not the bad one. It will confuse the user.

Reply:
I think it's ok.
f.e. if currentVersion is empty (which I dont expect that!), it will enter the first part of fold:
(s"current: $current")
and prints the value (current) that let to "None" in currentVersion